cmd/unifibackup/main.go

```diff
@@ -32,73 +32,56 @@ var (
+	} `cmd:"" help:"Generate autobackup_meta.json for the autobackup directory"`
+	Version kong.VersionFlag `env:"-"`
+	Backup  struct {
+		Dir     string        `help:"path of the UniFi autobackup directory." type:"existingdir" default:"/var/lib/unifi/backup/autobackup"`
+		Bucket  string        `help:"name of the S3 bucket to upload to." required:""`
+		Prefix  string        `help:"prepended to the backup file name to form the object key." default:"unifi/"`
+		Metrics string        `help:"listen spec for the web server that exposes Prometheus metrics." default:":9184"`
+		Timeout time.Duration `help:"time to allow for upload and delete requests." default:"5m"`
+	} `cmd:"" help:"Copies UniFi Controller backups to S3" default:"withargs"`
+}
+
 func main() {
 	maxprocs.Set() // use the library this way to avoid logging when CPU quota is undefined
 	buildInfo.WithLabelValues(stamp.Version, stamp.Commit).Set(1)
 	buildTime.Set(float64(stamp.Time().UnixNano()) / float64(time.Second))
 
-	// we can access flags via cmd, but they are untyped; in practice it's
-	// easier to constrain the chaos to this function
-	var (
-		flgBackupDir string
-		flgBucket    string
-		flgPrefix    string
-		flgMetrics   string
-		flgTimeout   time.Duration
-	)
-
-	rootCmd := &cobra.Command{
-		Use:          "unifibackup --bucket unifi-backups",
-		Short:        "Copies UniFi Controller backups to S3",
-		Version:      stamp.Summary(),
-		SilenceUsage: true,
-		Args:         cobra.NoArgs,
-		RunE: func(cmd *cobra.Command, _ []string) error {
-			ctx := cmd.Context()
-			cfg, err := config.LoadDefaultConfig(ctx,
-				config.WithUseDualStackEndpoint(aws.DualStackEndpointStateEnabled))
-			if err != nil {
-				return fmt.Errorf("failed to initialise AWS SDK: %w", err)
-			}
-			s3client := s3.NewFromConfig(cfg)
-			uploader := uploader.New(s3client, flgBucket, flgPrefix)
-			return daemon(ctx, flgMetrics, flgBackupDir, uploader, flgTimeout)
-		},
-	}
-	rootCmd.SetVersionTemplate("{{.Version}}\n")
-	rootCmd.PersistentFlags().StringVar(&flgBackupDir,
-		"dir",
-		"/var/lib/unifi/backup/autobackup",
-		"path of the UniFi autobackup directory")
-	rootCmd.Flags().StringVar(&flgBucket,
-		"bucket",
-		"", // in hindsight, this should not have been a flag
-		"name of the S3 bucket to upload to (requird)")
-	rootCmd.MarkFlagRequired("bucket")
-	rootCmd.Flags().StringVar(&flgPrefix,
-		"prefix",
-		"unifi/",
-		"prepended to the backup file name to form the object key")
-	rootCmd.Flags().StringVar(&flgMetrics,
-		"metrics",
-		":9184",
-		"listen spec for the web server that exposes Prometheus metrics")
-	rootCmd.Flags().DurationVar(&flgTimeout,
-		"timeout",
-		5*time.Minute,
-		"time to allow for upload and delete requests")
-
-	rootCmd.AddCommand(&cobra.Command{
-		Use:   "genmeta",
-		Short: "Generate autobackup_meta.json for the autobackup directory",
-		Args:  cobra.NoArgs,
-		RunE: func(_ *cobra.Command, _ []string) error {
-			return genmeta(flgBackupDir)
+	ctx := kong.Parse(
```

Comment:
`ctx` is conventionally reserved in Go for `context.Context`, and there is some shadowing going on as a result. How about `kongCtx`?